src/clang_utils.cc

```diff
@@ -20,30 +20,49 @@ lsRange GetLsRangeForFixIt(const CXSourceRange& range) {
 
 }  // namespace
 
+// See clang_formatDiagnostic
 optional<lsDiagnostic> BuildAndDisposeDiagnostic(CXDiagnostic diagnostic,
                                                  const std::string& path) {
   // Get diagnostic location.
   CXFile file;
-  unsigned int line, column;
-  clang_getSpellingLocation(clang_getDiagnosticLocation(diagnostic), &file,
-                            &line, &column, nullptr);
+  unsigned start_line, start_column, end_line, end_column;
+  clang_getSpellingLocation(clang_getDiagnosticLocation(diagnostic),
+                            &file, &start_line, &start_column, nullptr);
 
-  // Only report diagnostics in the same file. Using
+  // TODO Only report diagnostics in the same file. Using
```

Comment:
Why the TODO?

Reply:
Changed to 

```c++
  // Only report diagnostics in the same file. Using
  // clang_Location_isInSystemHeader causes crashes for some reason.
  // TODO investigate
```

The isInSystemHeader check may not be very appropriate.